### PR TITLE
Docsp 30927 work with bson options

### PR DESCRIPTION
**Pull request by nickldp**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-golang/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-30927>
[Staging](https://docs-mongodbcom-staging.corp.mongodb.com/golang/docsworker-xlarge/DOCSP-30927-Work-With-BSON-options/fundamentals/bson/) 

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [ ] Did you run a grammar-check?
- [x] Are all the links working?

### CODE REVIEW

File: source/fundamentals/bson.txt [hunk 1]
```diff
@@ -210,6 +210,29 @@ Driver will marshal structs using the following rules:
       - Includes an empty ``lastname`` field
       - Stores the ``Address`` field as a nested value
 
+BSON Options
+------------
+
+You can set ``BSONOptions`` to specify how
+the driver marshals and unmarshals BSON on any level, from
+``collection`` to ``Client``.
```
Comment (mongoKart) [inline source/fundamentals/bson.txt:218]: Issue: I'm not sure what "any level" refers to. How many are there?

Suggestion: Can we be more specific about these levels?

File: source/fundamentals/bson.txt [hunk 2]
```diff
@@ -210,6 +210,29 @@ Driver will marshal structs using the following rules:
...
+
+The following example demonstrates how the ``BSONOptions`` is set using
+the ``Movies`` collection as an example.
```
Comment (mongoKart) [inline source/fundamentals/bson.txt:221]: I: Two points from the Style Guide here: [Avoid passive voice](https://www.mongodb.com/docs/meta/style-guide/writing/use-active-voice/) and [Don't use "using" alone](https://www.mongodb.com/docs/meta/style-guide/terminology/alphabetical-terms/#u:~:text=word%2C%20username.-,using,-by%20using%20with)

S:

```suggestion
The following example demonstrates how to set ``BSONOptions`` for 
the ``Movies`` collection:
```

Comment (mongoKart) [inline source/fundamentals/bson.txt:221]: I: In our other docs, we keep collection names lowercase (e.g. [here](https://www.mongodb.com/docs/drivers/node/current/fundamentals/aggregation/#runnable-examples))

S:

```suggestion
The following example demonstrates how the ``BSONOptions`` is set using
the ``movies`` collection as an example.
```

File: source/fundamentals/bson.txt [hunk 3]
```diff
@@ -210,6 +210,29 @@ Driver will marshal structs using the following rules:
...
+
+   bsonOpts := &options.BSONOptions{
+		UseJSONStructTags: true,
+	}
+	coll := client.Database("sample_mflix").Collection("movies",
+	bsonOpts)
+   
+To learn more about the options and usage of ``BSONOptions``, visit the `API Documentation
+<https://pkg.go.dev/go.mongodb.org/mongo-driver/mongo#example-Connect-BSONOptions>`_
+for more information.
```
Comment (mongoKart) [inline source/fundamentals/bson.txt:234]: I: This sentence can probably be shorter and a little clearer

S:

```suggestion
To learn more about ``BSONOptions``, see the `API Documentation
<https://pkg.go.dev/go.mongodb.org/mongo-driver/mongo#example-Connect-BSONOptions>`__.
```

File: source/fundamentals/bson.txt [hunk 4]
```diff
@@ -210,6 +210,29 @@ Driver will marshal structs using the following rules:
...
+.. code-block:: go
+   :copyable: false
+
+   bsonOpts := &options.BSONOptions{
+		UseJSONStructTags: true,
+	}
+	coll := client.Database("sample_mflix").Collection("movies",
+	bsonOpts)
+   
+To learn more about the options and usage of ``BSONOptions``, visit the `API Documentation
```
Comment (mongoKart) [inline source/fundamentals/bson.txt:232]: S: Put this sentence in a Tip box, since it's more supplementary than a natural conclusion to this section

**Review by mongoKart — CHANGES_REQUESTED** [review]
Nice job! A few small changes

File: source/fundamentals/bson.txt [hunk 5]
```diff
@@ -210,6 +210,29 @@ Driver will marshal structs using the following rules:
...
+the ``Movies`` collection as an example.
+
+.. code-block:: go
+   :copyable: false
+
+   bsonOpts := &options.BSONOptions{
+		UseJSONStructTags: true,
+	}
+	coll := client.Database("sample_mflix").Collection("movies",
+	bsonOpts)
```
Comment (mongoKart) [inline source/fundamentals/bson.txt:230]: I: I think these lines should be dedented back to the beginning...

```suggestion
    coll := client.Database("sample_mflix").Collection("movies",
    bsonOpts)
```

File: source/fundamentals/bson.txt [hunk 6]
```diff
@@ -210,6 +210,29 @@ Driver will marshal structs using the following rules:
...
+   
+To learn more about the options and usage of ``BSONOptions``, visit the `API Documentation
+<https://pkg.go.dev/go.mongodb.org/mongo-driver/mongo#example-Connect-BSONOptions>`_
```
Comment (mongoKart) [inline source/fundamentals/bson.txt:233]: S: We almost always use source constants (in snooty.toml) to represent API links, making it easier and less error-prone. I would try using a source constant here.

File: source/fundamentals/bson.txt [hunk 7]
```diff
@@ -210,6 +210,30 @@ Driver will marshal structs using the following rules:
...
+the ``movies`` collection:
+
+.. code-block:: go
+   :copyable: false
+
+   bsonOpts := &options.BSONOptions{
+      UseJSONStructTags: true,
+	}
+   coll := client.Database("sample_mflix").Collection("movies",
+   bsonOpts)
```
Comment (mongoKart) [inline source/fundamentals/bson.txt:230]: S: I'm not a Go expert, but structuring the code like the following seems more readable:

```suggestion
   bsonOpts := &options.BSONOptions {
       UseJSONStructTags: true
   }
   coll := client.Database("sample_mflix")
       .Collection("movies", bsonOpts)
```
What do you think?

**Review by mongoKart — APPROVED** [review]
LGTM + 1 suggestion. Great job addressing the feedback!

File: source/fundamentals/bson.txt (hunk 7 above)
Comment (nickldp) [reply source/fundamentals/bson.txt:230]: Agree with everything, but comma on line 227 is needed

**Pull request merged**